### PR TITLE
Fix the build for classic templates.

### PR DESCRIPTION
**Pull request by singhsarab**

### CODE REVIEW

File: Templates/CSharp/Desktop/MSTestDesktopCSharp.csproj [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
-<Project ToolsVersion="15.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+﻿<Project  Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <TestTemplatesRoot Condition="$(TestTemplatesRoot) == ''">$(MSBuildThisFileDirectory)..\..\..\</TestTemplatesRoot>
   </PropertyGroup>
-  <Import Project="$(TestTemplatesRoot)scripts\build\TestTemplates.Settings.targets" />
+  <!--Import Project="$(TestTemplatesRoot)scripts\build\TestTemplates.Settings.targets" /-->
   <PropertyGroup>
+    <TargetFramework>net461</TargetFramework>
+    <OutputPath Condition="$(OutputPath) == ''">$(TestTemplatesRoot)artifacts\$(Configuration)\$(MSBuildProjectName)\</OutputPath>
```
Comment (chcosta) [inline Templates/CSharp/Desktop/MSTestDesktopCSharp.csproj]: I'd expect Arcade to define paths, this shouldn't be needed if you're using Arcade

File: Templates/CSharp/Desktop/MSTestDesktopCSharp.csproj [hunk 2]
```diff
@@ -29,6 +30,7 @@
     <CopyOutputSymbolsToOutputDirectory>false</CopyOutputSymbolsToOutputDirectory>
     <IsVsixProj>true</IsVsixProj>
```
Comment (chcosta) [inline Templates/CSharp/Desktop/MSTestDesktopCSharp.csproj]: In Arcade, the property appears to be `IsVsixProject`

https://github.com/dotnet/arcade/blob/9104c21492e1c6ed98105dd87388bca8f1f1abaa/src/Microsoft.DotNet.Arcade.Sdk/tools/VisualStudio.targets#L7

File: Templates/CSharp/Desktop/MSTestDesktopCSharp.csproj (hunk 1 above)
Comment (tmat) [reply Templates/CSharp/Desktop/MSTestDesktopCSharp.csproj]: Yes, don't define these paths.

File: Templates/CSharp/Desktop/MSTestDesktopCSharp.csproj [hunk 3]
```diff
@@ -1,10 +1,12 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
-<Project ToolsVersion="15.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+﻿<Project  Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <TestTemplatesRoot Condition="$(TestTemplatesRoot) == ''">$(MSBuildThisFileDirectory)..\..\..\</TestTemplatesRoot>
```
Comment (tmat) [inline Templates/CSharp/Desktop/MSTestDesktopCSharp.csproj]: Arcade defines `RepoRoot` property, so you don't need this one.

File: Templates/CSharp/Desktop/MSTestDesktopCSharp.csproj [hunk 4]
```diff
@@ -1,10 +1,12 @@
...
   </PropertyGroup>
-  <Import Project="$(TestTemplatesRoot)scripts\build\TestTemplates.Settings.targets" />
+  <!--Import Project="$(TestTemplatesRoot)scripts\build\TestTemplates.Settings.targets" /-->
   <PropertyGroup>
+    <TargetFramework>net461</TargetFramework>
+    <OutputPath Condition="$(OutputPath) == ''">$(TestTemplatesRoot)artifacts\$(Configuration)\$(MSBuildProjectName)\</OutputPath>
+    <IntermediatePath Condition="$(IntermediatePath) == ''">$(TestTemplatesRoot)artifacts\$(Configuration)\$(MSBuildProjectName)\obj\</IntermediatePath>
     <MinimumVisualStudioVersion>15.0</MinimumVisualStudioVersion>
     <VSToolsPath Condition="'$(VSToolsPath)' == ''">$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)</VSToolsPath>
```
Comment (tmat) [inline Templates/CSharp/Desktop/MSTestDesktopCSharp.csproj]: `VSToolsPath` shouldn't be needed.

File: Templates/CSharp/Desktop/MSTestDesktopCSharp.csproj [hunk 5]
```diff
@@ -29,6 +30,7 @@
     <CopyOutputSymbolsToOutputDirectory>false</CopyOutputSymbolsToOutputDirectory>
     <IsVsixProj>true</IsVsixProj>
     <DeployExtension>False</DeployExtension>
+	<EnableDefaultCompileItems>false</EnableDefaultCompileItems>
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
```
Comment (tmat) [inline Templates/CSharp/Desktop/MSTestDesktopCSharp.csproj]: Remove this PropertyGroup.

File: Templates/CSharp/Desktop/MSTestDesktopCSharp.csproj [hunk 6]
```diff
@@ -46,13 +48,16 @@
     <WarningLevel>4</WarningLevel>
   </PropertyGroup>
   <ItemGroup>
-    <Compile Include="Properties\AssemblyInfo.cs" />
+    <PackageReference Include="MicroBuild.Core" Version="0.2.0" />
```
Comment (tmat) [inline Templates/CSharp/Desktop/MSTestDesktopCSharp.csproj]: Don't reference MicroBuild or Microsoft.VSSDK.BuildTools

File: Templates/CSharp/Desktop/MSTestDesktopCSharp.csproj [hunk 7]
```diff
@@ -71,14 +76,15 @@
       <IncludeInVSIX>true</IncludeInVSIX>
     </Content>
   </ItemGroup>
-  <Import Project="$(TestTemplatesRoot)scripts\build\TestTemplates.targets" />
+  <!--Import Project="$(TestTemplatesRoot)scripts\build\TestTemplates.targets" /-->
   <PropertyGroup>
     <PreBuildEvent>
```
Comment (tmat) [inline Templates/CSharp/Desktop/MSTestDesktopCSharp.csproj]: What is this `<PreBuildEvent>` needed for?

File: Templates/CSharp/Desktop/MSTestDesktopCSharp.csproj (hunk 2 above)
Comment (singhsarab) [reply Templates/CSharp/Desktop/MSTestDesktopCSharp.csproj]: As per this, I am guessing we don't even need that https://github.com/dotnet/arcade/blob/9104c21492e1c6ed98105dd87388bca8f1f1abaa/src/Microsoft.DotNet.Arcade.Sdk/tools/VisualStudio.props#L14

**Comment by singhsarab** [comment]
@tmat @chcosta I have reacted based on the changes you have suggested. Now I see there is an output dll but no vsix as such for this project. What do you think we are missing here ?

**Comment by tmat** [comment]
@singhsarab Can you please share binlog?

Run 
```
build -bl
```

and share `artifacts\log\Debug\Build.binlog` that's produced by the build.

**Comment by tmat** [comment]
Never mind, I think I see what's wrong.

Try to add 
`<UsingToolVSSDK>true</UsingToolVSSDK>`
here:
https://github.com/dotnet/test-templates/blob/arcade/eng/Versions.props#L5

File: Templates/CSharp/Desktop/MSTestDesktopCSharp.csproj [hunk 8]
```diff
@@ -63,22 +30,25 @@
     <Content Include="packages\MSTest.TestFramework.1.4.0.nupkg">
       <IncludeInVSIX>true</IncludeInVSIX>
     </Content>
-    <None Include="source.extension.vsixmanifest">
+    <!--None Include="source.extension.vsixmanifest">
```
Comment (tmat) [inline Templates/CSharp/Desktop/MSTestDesktopCSharp.csproj]: The manifest needs to be included.

**Pull request merged**